### PR TITLE
Add other topics to topic "hacktoberfest"

### PR DESCRIPTION
**Pull request by shaedrich**
### Please confirm this pull request meets the following requirements:

- [x] I followed the contributing guidelines: <https://github.com/github/explore/blob/main/CONTRIBUTING.md>.
- [x] I am not the sole author or employee of a company who created the topic or collection I'm changing.

### Which change are you proposing?

  - [x] Suggesting edits to an existing topic or collection
  - [ ] Curating a new topic or collection
  - [ ] Something that does not neatly fit into the binary options above

---

### Editing an existing topic or collection

I'm suggesting these edits to an existing topic or collection:
- [ ] Image (and my file is `*.png`, square, dimensions 288x288, size <= 75 kB)
- [x] Content (and my changes are in `index.md`)

There are currently almost countless topics and since Hacktoberfest is important to GitHub and many developers all around the globe, this is a way to find a fitting topic and contribute to it.

### Curating a new topic or collection

- [ ] I've formatted my changes as a new folder directory, named for the topic or collection as it appears in the URL on GitHub (e.g. `https://github.com/topics/[NAME]` or `https://github.com/collections/[NAME]`)
- [ ] My folder contains a `*.png` image (if applicable) and `index.md`
- [ ] All required fields in my `index.md` conform to the Style Guide and API docs: <https://github.com/github/explore/tree/main/docs>

> Please replace this line with an explanation of why you think this topic or collection should be curated.

### Something that does not neatly fit into the binary options above

- [ ] My suggested edits are not about an existing topic or collection, or at least not a single one
- [ ] My suggested edits are not about curating a new topic or collection, or at least not a single one
- [ ] My suggested edits conform to the Style Guide and API docs: https://github.com/github/explore/tree/main/docs

> Please replace this line with an explanation of your proposed changes.

---

**Please note: we will close your PR without comment if you do not check the boxes above and provide ALL requested information.**

### CODE REVIEW

**Review by octokatherine — APPROVED** [review]
Thanks for your contribution!

**Comment by shaedrich** [comment]
You're welcome!

**Comment by MattIPv4** [comment]
👋 @shaedrich @octokatherine I'm a bit concerned that this change is going to confuse Hacktoberfest users? Especially the "status" topics that're being included here. None of these topics are endorsed by Hacktoberfest, nor do any of them indicate a repository is participating in Hacktoberfest. A repository that is using `hacktoberfest2023` is not opted in to Hacktoberfest, and PRs there won't count, they need the `hacktoberfest` topic. Similarly, a repository with `hacktoberfest-accepted`/`hacktoberfest-approved` is not opted in, and those topics carry no weight/endorsement.

**Comment by shaedrich** [comment]
Thanks for your perspective on this. To clear up the confusion: I checked other topics containing "hacktoberfest" and thought about adding them as aliases. However, both `aliases` and `related` can only have a limited number of items. Therefore, I thought, to properly use topics to their full potential, there are only two courses of action, on can take on a topic, benefitting other topics taking the other course of action:
* Stop using a topics that makes less sense than a similar topic, so that it becomes entirely unused
* Use a better topic instead to make it more popular and therefore more commonly accepted

So I suppose, since people are using these topics, people need to know, they exist (or being offered alternatives) to streamline the huge variety of topics.

In conclusion: What you say, makes sense so far. However, I don't think, it's good to act as if these topics don't exist. But I'm not sure, how to go from here. Notifying each and every repository owner/maintainer that there might be better topic choices manually doesn't sound too convincing. Does what I say make sense?

**Comment by MattIPv4** [comment]
I understand what you're saying, and that folks are already using some of these topics -- most of the topics you've introduced in this PR are really not being used and return less than 5 repositories each. Only the year-based topics seem to be more heavily used.

However, as I noted, none of these additional topics are tracked for Hacktoberfest, Hacktoberfest only considers the `hacktoberfest` topic for repositories to opt-in, so encouraging the use of these additional topics by linking out to them on the page is only going to lead to further confusion and "incorrect" usage.

**Comment by shaedrich** [comment]
I understand that so far. But that doesn't really solve the problem of creating clutter by assigning repositories to these topics but not being able to list each and every one as alias so they are taken care of. Maybe I want this to be too cleaned up 😅

**Comment by MattIPv4** [comment]
Yeah, it'd be great to have more as aliases, but honestly I don't think it's worth worrying about really. Making them more visible by listing them on the topic page certainly does not help with reducing their usage and cleaning them up, it'll just encourage more usage of them, creating more clutter.

**Comment by MattIPv4** [comment]
I left a comment in https://github.com/github/explore/pull/4042#issuecomment-1810581332 that touches on what I think would be useful -- essentially guidance to help new users on discovering issues that they could tackle to contribute to Hacktoberfest projects. GitHub doesn't support search features to enable that though.

Beyond that though, I don't think there is really a need to put a large amount of information on the topic page? I would imagine most folks that're looking at the Hacktoberfest topic already know what Hacktoberfest is, and for those that don't, there is already a succinct summary of what it is, and a link out to the site will all the information.

**Comment by shaedrich** [comment]
> However, I don't think, it's good to act as if these topics don't exist. But I'm not sure, how to go from here. Notifying each and every repository owner/maintainer that there might be better topic choices manually doesn't sound too convincing. Does what I say make sense?

Maybe, as this is more of a general question, I redirect the question to @octokatherine.
